recipes/bitmagic/all/test_package/conanfile.py

```diff
@@ -8,6 +8,10 @@ class TestPackageConan(ConanFile):
 
     def build(self):
         cmake = CMake(self)
+        if tools.Version(self.deps_cpp_info["bitmagic"].version) < "7.5.0":
+            cmake.definitions["CMAKE_CXX_STANDARD"] = 11
+        else:
+            cmake.definitions["CMAKE_CXX_STANDARD"] = 17
```

Comment:
I would enclose this into an `if` to check if the user is providing the standard from outside, something like:

```suggestion
        if not self.settings.compiler.get_safe("cppstd"):
            if tools.Version(self.deps_cpp_info["bitmagic"].version) < "7.5.0":
                cmake.definitions["CMAKE_CXX_STANDARD"] = 11
            else:
                cmake.definitions["CMAKE_CXX_STANDARD"] = 17
```